lib/BackgroundJob/ConvertViewColumnsFormat.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OCA\Tables\BackgroundJob;
+
+use OCP\AppFramework\Utility\ITimeFactory;
+use OCP\BackgroundJob\IJobList;
+use OCP\BackgroundJob\Job;
+use OCP\IDBConnection;
+use OCP\IUserManager;
+
+class ConvertViewColumnsFormat extends Job {
+	public function __construct(
+		ITimeFactory $time,
+		protected IDBConnection $connection,
+		protected IUserManager $userManager,
+	) {
+		parent::__construct($time);
+	}
+
+	public function run($argument) {
+		$qb = $this->connection->getQueryBuilder();
+		
+		// Get all views that need processing
+		$qb->select('id', 'columns')
+			->from('tables_views')
+			->where($qb->expr()->isNotNull('columns'));
+
+		$result = $qb->executeQuery();
+		
+		// Predefine update query
+		$updateQb = $this->connection->getQueryBuilder();
+		$updateQb->update('tables_views')
+			->set('columns', $updateQb->createParameter('columns'))
+			->where($updateQb->expr()->eq('id', $updateQb->createParameter('id')));
+
+		$maxId = 0;
+		// Update each view
+		while ($view = $result->fetch()) {
+			$this->processView($view, $updateQb);
+			$maxId = max($maxId, (int)$view['id']);
+		}
+
+		$result->closeCursor();
+
+		// Check for new views added during processing
+		$qb = $this->connection->getQueryBuilder();
+		$qb->select('id', 'columns')
+			->from('tables_views')
+			->where($qb->expr()->gt('id', $qb->createNamedParameter($maxId, \PDO::PARAM_INT)))
+			->andWhere($qb->expr()->isNotNull('columns'));
+
+		$result = $qb->executeQuery();
+
+		// Process new views if any
+		while ($view = $result->fetch()) {
+			$this->processView($view, $updateQb);
+		}
+
+		$result->closeCursor();
+	}
+
```

Comment:
for this should be run once only, the job has to be removed from the list, as in
 
https://github.com/nextcloud/server/blob/a8f46af20f4fccac0257eba950e70d0da96c4a5a/lib/public/BackgroundJob/QueuedJob.php#L37-L43